src/Metadata/Resource/Factory/LinkFactory.php

```diff
@@ -59,6 +59,9 @@ public function createLinksFromIdentifiers(Metadata $operation): array
 
         $link = (new Link())->withFromClass($resourceClass)->withIdentifiers($identifiers);
         $parameterName = $identifiers[0];
+        if ('value' === $parameterName && enum_exists($resourceClass)) {
+            $parameterName = 'id';
+        }
```

Comment:
Does this match what you had in mind @soyuka?

Reply:
Uri variable should be as followed:

```
uriVariables: ['id' => new Link(parameterName: 'id', identifiers: ['value'])]
```

in the metadata directly so that we don't need any condition here right?

Reply:
Apologies, but I'm confused again :innocent: … I'm probably missing something obvious, but the `Link` is constructed directly above, and the `parameterName` is set at return: `$link->withParameterName($parameterName)` … So is you suggesting how that return array should be shaped? If so what about non-enums?

Reply:
Which raises another thing about the `createLinksFromIdentifiers` method … If the `$identifier` array is empty there is an early return (lines 56-58), and then there is the `if (1 < \count($identifiers)) {` (lines 66-69) test that looks like dead code, correct?

Reply:
I need to check this into the details, will try to find some time!

Reply:
okay I know understand, basically what I had in bind was

```
            if (enum_exists($operation->getClass())) {
                $operation = $operation->withUriVariables(['id' => new Link(parameterName: 'id', identifiers: ['value'])]);
            } else {
```

directly inside the `UriTemplateResourceMetadataCollectionFactory` class, your version also populates a Link that didn't specified any identifiers as materialized by the test at `testEnumsAreAssignedValuePropertyAsIdentifierByDefault` ! Thanks!